File: python/snake_game/audio_manager.py
import pygame


class AudioManager:
    def __init__(self, asset_dir='assets'):
        self.enabled = True
        try:
            self._eat = pygame.mixer.Sound(f'{asset_dir}/eating sound effect.wav')
            self._death = pygame.mixer.Sound(f'{asset_dir}/death sound.wav')
        except pygame.error:
            # no audio device (headless/CI), fail soft
            self.enabled = False
            self._eat = None
            self._death = None

    def play_eat(self):
        if self.enabled:
            self._eat.play()

    def play_death(self):
        if self.enabled:
            self._death.play()

File: cpp/src/audio_manager.cpp
#include "audio_manager.hpp"

AudioManager::AudioManager() {
    InitAudioDevice();
    // no audio device (headless/CI), fail soft
    enabled_ = IsAudioDeviceReady();
    if (enabled_) {
        eat_ = LoadSound("assets/eating sound effect.wav");
        death_ = LoadSound("assets/death sound.wav");
        enabled_ = eat_.frameCount > 0 && death_.frameCount > 0;
    }
}

AudioManager::~AudioManager() {
    if (enabled_) {
        UnloadSound(eat_);
        UnloadSound(death_);
    }
    if (IsAudioDeviceReady()) CloseAudioDevice();
}

void AudioManager::PlayEat() const {
    if (enabled_) PlaySound(eat_);
}

void AudioManager::PlayDeath() const {
    if (enabled_) PlaySound(death_);
}
